packages/scene-composer/src/components/panels/CustomColorPicker/CustomColorPickerHelpers.tsx

```diff
@@ -0,0 +1,133 @@
+import { Input, InputProps } from '@awsui/components-react';
+import React, { forwardRef } from 'react';
+import styled from 'styled-components';
+import * as awsui from '@awsui/design-tokens';
+
+export type RGB = {
+  r: string;
+  g: string;
+  b: string;
+};
+
+/**
+ * Converts a hex color to rgb format
+ *
+ * @param color - Either in form '#XXXXXX' or '#XXX'
+ * @returns The rgb version of the color
+ */
+export const hexToRgb = (color: string): RGB => {
+  if (color.length === 7) {
+    return {
+      r: parseInt(color.substring(1, 3), 16).toString(),
+      g: parseInt(color.substring(3, 5), 16).toString(),
+      b: parseInt(color.substring(5, 7), 16).toString(),
+    };
+  } else {
+    return {
+      r: parseInt(color.at(1)! + color.at(1)!, 16).toString(),
+      g: parseInt(color.at(2)! + color.at(2)!, 16).toString(),
+      b: parseInt(color.at(3)! + color.at(3)!, 16).toString(),
+    };
+  }
+};
+
+/**
+ * Formats a string to look like a hex value.
+ *
+ * @param value - Any string
+ * @returns A string starting with '#' and with all non-hex values removed
+ */
+export const validateHex = (value: string) => {
+  return `#${value.replace(/[^a-fA-F0-9]/g, '').substring(0, 6)}`;
+};
+
+/**
+ * Checks if a given hex code is valid
+ *
+ * @param value - Any string
+ * @returns True if value is of form '#XXXXXX' or '#XXX', false otherwise
+ */
+export const isValidHex = (value: string) => {
+  const validHexRegex = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;
+  return validHexRegex.test(value);
+};
+
+/**
+ * Formats a string to look like a rgb value
+ *
+ * @param value - Any string
+ * @returns A string containing a number between 0-255 with no leading 0's (unless it would be blank otherwise)
+ */
+export const validateRgbValue = (value: string) => {
+  const onlyDigits = value.replace(/\D/g, '');
+  return parseInt(onlyDigits) > 255 ? '255' : parseInt(onlyDigits.padStart(1, '0')).toString();
+};
+
+//Converts a string in decimal form to a string in hex with two digits
+/**
+ * Formats a number in a string to hexidemal in a string
+ *
+ * @param value - A string containing only numeric characters
+ * @returns A string containing the hexidecimal equivalent of the input
+ */
+const decimalToHex = (value: string) => {
```

Comment:
A few cases to consider,
1. What happens when we pass negative numbers?
2. What happens when pass large number?

Reply:
As it stands, this function is only called by RgbToHex (which I will rename to rgbToHex), which itself is only called after the rgb input has been validated by validateRgbValue. This ensures that decimalToHex will only be called with a string containing a number between 0-255. However this implementation may change in the future, and your questions could lead to some bad errors. 

Do you think updating the comments above the function to say that the input must be a string of a positive number between 0-255 is enough? Or would it be best if I added some checks to the function?

Reply:
I didn't know about three's `ColorRepresentation` but it looks really clean 😀. Will switch over to that